pkg/proxy/userspace/proxier.go

```diff
@@ -287,6 +290,20 @@ func CleanupLeftovers(ipt iptables.Interface) (encounteredError bool) {
 	return encounteredError
 }
 
+// Shutdown closes all service port proxies and returns from the proxy's
+// sync loop
+func (proxier *Proxier) Shutdown() {
```

Comment:
I'd make this internal (ie, `shutdown()`) and just for testing. Otherwise you have to prove to me that dropping the mutex temporarily partway through the shutdown (between `close(proxier.stopChan)` and `proxier.cleanupStaleStickySessions()`) can't result in race conditions in any possible scenario.

Reply:
The locking-in-the-middle is only used in this interim commit; that gets fixed later when we remove the locking from cleanupStaleStickySessions(). But changed anyway since nothing other than testcsaes uses it.